docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,778 @@
+====
+
+Now you are ready to start testing the endpoint.
+
+== Test Quarkus endpoint - first try
+
+Start Quarkus in `devmode`:
+
+[source,bash]
+----
+$ mvn quarkus:dev
+----
+
+This is the only time you are expected to start Quarkus manually in devmode while working with this tutorial. All the configuration and code updates in the follow up sections is expected to be recognized by Quarkus without having to restart the application manually.
+
+Open the browser and access `http://localhost:8080/hello`, you will be redirected to `Auth0` where you will login:
+
+image::auth0-login.png[Auth0 Login]
+
+and authorize the `QuarkusAuth0` application to access your account:
+
+image::auth0-authorize.png[Auth0 Authorize]
+
+and finally you will be redirected back to the Quarkus endpoint and get a response, `Hello, auth0|60e5a305e8da5a006aef5471`.
+
+This is not the current user's name though, so why has this strange sequence been returned ?
+
+Lets use https://quarkus.io/guides/security-openid-connect-dev-services#dev-ui-all-oidc-providers[Quarkus OIDC Dev UI console] to find out.
+
+== Looking at Auth0 tokens in OIDC DevUI
+
+Quarkus provides a great https://quarkus.io/guides/dev-ui-v2[DevUI] experience. In particular, it offers an out of the box support for developing and testing OIDC endpoints with a Keycloak container, https://quarkus.io/guides/security-openid-connect-dev-services#dev-services-for-keycloak[DevService for Keycloak] which is launched when the OIDC provider's address is not already configured with `quarkus.oidc.auth-server-url`.
+
+However, Quarkus users can continue using OIDC DevUI when the provider is already configured. Lets check it right now.
+
+First though, make sure you change your Quarkus application to have a `hybrid` application type, instead of the current `web-app`:
+
+[source,configuration]
+----
+quarkus.oidc.auth-server-url=https://dev-3ve0cgn7.us.auth0.com
+quarkus.oidc.application-type=hybrid <1>
+quarkus.oidc.client-id=sKQu1dXjHB6r0sra0Y1YCqBZKWXqCkly
+quarkus.oidc.credentials.secret=${client-secret}
+----
+<1> Application type is changed to `hybrid`. The reason it was changed to `hybrid` is because OIDC Dev UI currently supports an `SPA` mode only, i.e, it itself, using its own Java Script, authenticates users to the OIDC provider, and uses the access token as a `Bearer` token to access the Quarkus endpoint as a service.
```

Comment:
```suggestion
<1> Application type is changed to `hybrid` because OIDC Dev UI currently supports `SPA` mode only.
For example, using its own Java Script, the application authenticates users to the OIDC provider and uses the access token as a Bearer token to access the Quarkus endpoint as a service.
```

Reply:
@michelle-purcell It makes it not quite obvious which application uses `JavaScript`, let me tweak it a bit